src/main/java/tan/TaskList.java

```diff
@@ -0,0 +1,289 @@
+package tan;
+
+import tan.exceptions.DukeFormatExceptions;
+import tan.tasktype.Deadline;
+import tan.tasktype.Event;
+import tan.tasktype.Task;
+import tan.tasktype.ToDo;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TaskList {
+
+    protected static List<Task> listOfTasks = new LinkedList<>();
+    private static int totalNumberOfTask = 0;
+
+    /**
+     * Adds a new task by taking the user input as a string
+     * and decides what type of task it is and
+     * creates it. If none of the task type
+     * matches, prompts the user to try again.
+     *
+     * @param userInput The entire input from the user as a String.
+     */
+    public static void addTask(String userInput) {
+        String typeOfTask = Parser.getTypeOfTask(userInput);
+        Task curTask;
+        switch (typeOfTask) {
+        case "todo":
+            curTask = createTodoTask(userInput);
+            if (curTask == null) {
+                return;
+            }
+            break;
+        case "deadline":
+            curTask = createDeadlineTask(userInput);
+            if (curTask == null) {
+                return;
+            }
+            break;
+        case "event":
+            curTask = createEventTask(userInput);
+            if (curTask == null) {
+                return;
+            }
```

Comment:
There are repeated `curTask == null`-  how about checking `curTask` at the end of the switch statement? This is to improve the[ execution flow](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent)

Reply:
Thank you for pointing that out! I didn't notice it. I've changed it.